internal/jobworker/cli/cli.go

```diff
@@ -0,0 +1,98 @@
+// Package cli defines the jobworker CLI.
+package cli
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/tjper/teleport/internal/jobworker"
+	"github.com/tjper/teleport/internal/log"
+)
+
+var (
+	keyFlag    = flag.String("key", "", "path to server private key")
+	certFlag   = flag.String("cert", "", "path to server certificate")
+	caCertFlag = flag.String("ca_cert", "", "path to CA certificate")
+	portFlag   = flag.Int("port", 8080, "port to serve jobworker API")
+)
+
+// logger is an object for logging package events to stdout.
+var logger = log.New(os.Stdout, "cli")
+
+const (
+	ecSuccess = iota
+	// ecUnrecognized indicates the subcommand was not recognized.
+	ecUnrecognized
+	// ecCgroupService indicates the cgroup service was not setup properly.
+	ecCgroupService
+	// ecJobService indicates the job service was not setup properly.
+	ecJobService
+	// ecTLSConfig indicates the TLS config was not setup properly.
+	ecTLSConfig
+	// ecListen indicates the jobworker API was unable to listen.
+	ecListen
+	// ecServe indicates the jobworker API was unable to serve its content.
+	ecServe
+)
+
+const (
+	// serve is the subcommand used to serve the jobworker API.
+	serveSub = "serve"
+)
+
+// Run is the entrypoint of the jobworker CLI.
+func Run() int {
+	flag.Parse()
+
+	if len(os.Args) < 2 {
+		return help("Too few arguments")
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	last := len(os.Args) - 1
+	switch v := os.Args[last]; v {
+	case serveSub:
+		return runServe(ctx)
+	case jobworker.Reexec:
+		return runReexec(ctx)
+	default:
+		return help(fmt.Sprintf("Unrecognized subcommand \"%s\".", v))
+	}
+}
+
+// help outputs a general overview of the jobworker executable to the user.
+// The text argument may be used to add a detailed help message.
+func help(text string) int {
+	var b strings.Builder
+	if text != "" {
+		_, _ = b.WriteString(fmt.Sprintf("\nNotice: %s", text))
```

Comment:
```suggestion
		_, _ = fmt.Fprintf(b, "\nNotice: %s", text) 
```

`strings.Builder` implements `io.Writer` so this can be simplified.